tool/synthetic-data-sampler.ts

```diff
@@ -0,0 +1,530 @@
+// -*- mode: js; indent-tabs-mode: nil; js-basic-offset: 4 -*-
+//
+// This file is part of Genie
+//
+// Copyright 2021 The Board of Trustees of the Leland Stanford Junior University
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author: Jake Wu <jmhw0123@gmail.com>
+
+
+import * as argparse from 'argparse';
+import * as fs from 'fs';
+import * as Tp from 'thingpedia';
+import * as I18n from '../lib/i18n';
+import seedrandom from 'seedrandom';
+import { Ast } from 'thingtalk';
+import { Type, SchemaRetriever } from 'thingtalk';
+import { choose } from '../lib/utils/random';
+import { sampleString } from '../lib/utils/misc-utils';
+import { ParaphraseExample } from './autoqa/lib/canonical-example-constructor';
+import { generateExamples } from './autoqa/lib/canonical-example-constructor';
+import { parseConstantFile } from './lib/constant-file';
+import { getElementType } from './autoqa/wikidata/utils';
+import { clean } from '../lib/utils/misc-utils';
+import { makeLookupKeys } from '../lib/dataset-tools/mturk/sample-utils';
+import { PARTS_OF_SPEECH, Canonicals, CanonicalAnnotation } from './autoqa/lib/base-canonical-generator';
+import genBaseCanonical from './autoqa/lib/base-canonical-generator';
+import * as utils from '../lib/utils/misc-utils';
+import { serializePrediction } from '../lib/utils/thingtalk';
+import { EntityUtils } from '../lib';
+
+var path = require('path');
+
+export interface Entity {
+    value : string;
+    display : string;
+}
+
+interface Constant {
```

Comment:
are you trying to use this for string constant as well? this is confusing, `Constant` should probably be an `Entity` or a string.